### PR TITLE
Propose for benchmark testing

### PR DESCRIPTION
**Pull request by Dzalevski**
2 proposals
- Repetitive test 
- Test cases benchmark test

--ex from running the benchmark
``` terminal
go test  -bench=. | prettybench                                                                                   
goos: darwin
goarch: arm64
pkg: moul.io/u
PASS
benchmark                              iter    time/iter
---------                              ----    ---------
BenchmarkB64Encode/1000-8        1000000000   0.00 ns/op
BenchmarkB64Encode/100000-8      1000000000   0.00 ns/op
BenchmarkB64Encode/100000000-8   1000000000   0.07 ns/op
BenchmarkB64Decode/1000-8        1000000000   0.00 ns/op
BenchmarkB64Decode/100000-8      1000000000   0.00 ns/op
BenchmarkB64Decode/100000000-8   1000000000   0.05 ns/op
ok      moul.io/u       2.213s
```

### CODE REVIEW

File: encoding_test.go [hunk 1]
```diff
@@ -39,3 +41,20 @@ func ExampleJSON() {
...
+	}{
+		{DataSize: 1000},
+		{DataSize: 100000},
+		{DataSize: 100000000},
+	}
+	for _, bc := range cases {
+		b.Run(fmt.Sprintf("%d", bc.DataSize), func(b *testing.B) {
+			data := make([]byte, bc.DataSize)
+			rand.Read(data)
+			u.B64Encode(data)
```
Comment (moul) [inline encoding_test.go:57]: why not using `b.N` anymore? I think it was a good idea

**Review by moul — COMMENTED** [review]
a lot better, just two small things again

File: encoding_test.go [hunk 2]
```diff
@@ -39,3 +41,20 @@ func ExampleJSON() {
...
+		DataSize int
+	}{
+		{DataSize: 1000},
+		{DataSize: 100000},
+		{DataSize: 100000000},
+	}
+	for _, bc := range cases {
+		b.Run(fmt.Sprintf("%d", bc.DataSize), func(b *testing.B) {
+			data := make([]byte, bc.DataSize)
+			rand.Read(data)
```
[the remark this answers is not in the document]
Comment (Dzalevski) [reply encoding_test.go:56]: @moul  yes, you are right. we should use bytes.Repeat() -> it's the same outcome.

**Review by moul — APPROVED** [review]

**Comment by moul** [comment]
with fixed benchs

Screenshot - Manfred 2021-05-24 at 04 51 24

**Comment by moul** [comment]
And here, with parallelism

Screenshot - Manfred 2021-05-24 at 05 00 03

**Comment by moul** [comment]
@all-contributors please add @Dzalevski for code

**Pull request merged**